designs/degraded-nodepools.md

```diff
@@ -0,0 +1,119 @@
+# RFC: Degraded NodePool Status Condition
+
+## Overview
+
+Karpenter can launch nodes with a NodePool that will never join a cluster when a NodeClass is misconfigured.
+
+One example is that when a network path does not exist due to a misconfigured VPC (network access control lists, subnets, route tables), Karpenter will not be able to provision compute with that NodeClass that joins the cluster until the error is fixed. Crucially, this will continue to charge users for compute that can never be used in a cluster.
+
+To improve visibility of these failure modes, this RFC proposes adding a `Degraded` status condition on the Nodepool that indicate to cluster users there may be a problem with a NodePool/NodeClass combination that needs to be investigated and corrected. 
```

Comment:
I think it would also be worth giving a counter example for the types of errors we're not trying to capture with the deprecated condition:
> The degraded status condition should not be used as a catch-all for any launch / registration failures. If Karpenter can guarantee that a launch / registration will not succeed based on the information provided in the NodePool / NodeClass configs, it shouldn't consider those resources ready. For example, if a NodePool is constrained to a zone that is not available via the provided subnets, Karpenter should treat that NodePool as not ready rather than degraded.